client/my-sites/domains/domain-management/dns/dns-records-list.jsx

```diff
@@ -116,9 +115,8 @@ class DnsRecordsList extends Component {
 
 	deleteDns = ( record, action = 'delete', confirmed = false ) => {
 		const { selectedDomainName, translate } = this.props;
-		const { records } = this.props.dns;
 
-		if ( ! confirmed && isDeletingLastMXRecord( record, records ) ) {
+		if ( ! confirmed && record.protected_field ) {
```

Comment:
I know that at the moment the only protected deletable record is MX, but what about adding an extra check?

```suggestion
		if ( ! confirmed && record.protected_field  && 'MX' === record.type) {
```

Reply:
It wouldn't hurt 😛

Reply:
Cool! 😁